### PR TITLE
pass context to mapPropsToFields

### PR DESCRIPTION
**Pull request by plandem**
Close: https://github.com/react-component/form/issues/68

### CODE REVIEW

File: src/createBaseForm.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ function createBaseForm(option = {}, mixins = []) {
         };
       },
 
-      componentWillReceiveProps(nextProps) {
+      componentWillReceiveProps(nextProps, nextContext) {
         if (mapPropsToFields) {
-          this.fields = mapPropsToFields(nextProps);
+          this.fields = mapPropsToFields(nextProps, nextContext);
```
Comment (benjycui) [inline src/createBaseForm.js]: Sorry, I make a mistake.

It seems that you cannot get any value from `nextContext`, for this HOC doesn't subscribe any value, right?

Comment (benjycui) [inline src/createBaseForm.js]: I reverted this PR in master branch.

Comment (plandem) [inline src/createBaseForm.js]: It's same issue as with **connect** from redux. You have to defined contextTypes for wrapped component. E.g.:
```
class A extends React.Component {...}

A = Form.create(mapper)(A);
A.contextTypes = {
}
```

Comment (benjycui) [inline src/createBaseForm.js]: I think you can use a HOC to convert context to props, and then pass it to form.

Comment (benjycui) [inline src/createBaseForm.js]: https://github.com/react-component/form/pull/69#discussion_r112400778 this is not a solution, it is confusing.

**Comment by fancygeek** [comment]
any update?

**Pull request merged**